### PR TITLE
Oracle GoldenGate on Docker

### PR DESCRIPTION
**Pull request by sbalousek**
- Dockerfile and associated scripts for running Oracle GoldenGate in a Docker Container.

Signed-off-by: Stephen Balousek <stephen.balousek@oracle.com>

### CODE REVIEW

**Comment by Djelibeybi** [comment]
Do you have CorpArch approval to merge this code? Under that approval, it's likely the license will need to change to the UPL as well instead of GPL+CDDL.

If you do have CorpArch approval, please forward the approval email to @brunoborges and I internally. If not, please close this PR and only re-submit once you have CorpArch approval. 

This is a new requirement outside of our control.

**Comment by brunoborges** [comment]
@Djelibeybi this PR is approved. Let's review technically the content now.

**Comment by Djelibeybi** [comment]
Oh, terrific. Will do. :)

File: OracleGoldenGate/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# LICENSE CDDL 1.0 + GPL 2.0
```
Comment (brunoborges) [inline OracleGoldenGate/Dockerfile]: Please change the license to UPL

File: OracleGoldenGate/LICENSE [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Unless otherwise noted, all files in this folder are released
```
Comment (brunoborges) [inline OracleGoldenGate/LICENSE]: Approved license is UPL, not GPL.

File: OracleGoldenGate/dockerBuild.sh [hunk 3]
```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+# LICENSE CDDL 1.0 + GPL 2.0
```
Comment (brunoborges) [inline OracleGoldenGate/dockerBuild.sh]: Approved license is UPL. Update.

File: OracleGoldenGate/runOracleGoldenGate.sh [hunk 4]
```diff
@@ -0,0 +1,272 @@
+#!/bin/bash
+# LICENSE CDDL 1.0 + GPL 2.0
```
Comment (brunoborges) [inline OracleGoldenGate/runOracleGoldenGate.sh]: Approved license is UPL.

File: OracleGoldenGate/README.md [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+#### SSL Certificate for Microservices Architecture
+When the Oracle GoldenGate Docker image is created for Microservices Architecture, a dummy SSL certificate is generated for the OGG Web UI. Your own SSL certificate can be used instead of the dummy certificate like this:
+
+    docker run --name ogg-test \
+        -e OGG_SCHEMA=ggadmin \
+        -e OGG_ADMIN=oggadmin \
+        -v /path/to/certificate.pem:/etc/nginx/ogg.pem \
+        oracle/goldengate-microservices:12.3.0.1.0
+
+The certificate file, `/path/to/certificate.pem`, needs to contain a **BEGIN PRIVATE KEY** section, followed by one or more **BEGIN CERTIFICATE** sections. For example:
```
Comment (Djelibeybi) [inline OracleGoldenGate/README.md]: I would be more specific and state that you need a full chain certificate including the private key and all intermediate and root CA public keys.

**Comment by sbalousek** [comment]
I followed the guidelines at https://oss.oracle.com/licenses/upl/#_How_should_I, and they indicate that a full copy of the license is always preferred.

File: OracleGoldenGate/Dockerfile [hunk 6]
```diff
@@ -1,6 +1,32 @@
-# LICENSE CDDL 1.0 + GPL 2.0
+# Copyright (c) 2017 Oracle and/or its affiliates.  All rights reserved.
 #
-# Copyright (c) 2014-2017 Oracle and/or its affiliates. All rights reserved.
+# The Universal Permissive License (UPL), Version 1.0
```
Comment (brunoborges) [inline OracleGoldenGate/Dockerfile]: There is no need to paste the entire License here. Just say it is licensed under UPL 1.0

File: OracleGoldenGate/README.md [hunk 7]
```diff
@@ -228,9 +228,9 @@ The **Admin Client** utility can be run in the OGG container with this command:
 None
 
 ## License
-To download and run Oracle GoldenGate, regardless whether inside or outside a Docker container, you must download the binaries from the Oracle website and accept the license indicated at that page.
+All scripts and files hosted in this project and GitHub [docker-images/OracleGoldenGate](./) repository required to build the Docker images are, unless otherwise noted, released under the Universal Permissive License (UPL), Version 1.0.  See [LICENSE](./LICENSE) for details.
 
-All scripts and files hosted in this project and GitHub [docker-images/OracleGoldenGate](./) repository required to build the Docker images are, unless otherwise noted, released under the Common Development and Distribution License (CDDL) 1.0 and GNU Public License 2.0 licenses.
+To download and run Oracle GoldenGate, regardless whether inside or outside a Docker container, you must download the binaries from the Oracle website and accept the license indicated at that page.
```
Comment (brunoborges) [inline OracleGoldenGate/README.md]: Add the URL to where users must go to download the binary.

File: OracleGoldenGate/dockerBuild.sh [hunk 8]
```diff
@@ -1,7 +1,33 @@
 #!/bin/bash
-# LICENSE CDDL 1.0 + GPL 2.0
+# Copyright (c) 2017 Oracle and/or its affiliates.  All rights reserved.
 #
-# Copyright (c) 2014-2017 Oracle and/or its affiliates. All rights reserved.
+# The Universal Permissive License (UPL), Version 1.0
```
Comment (brunoborges) [inline OracleGoldenGate/dockerBuild.sh]: Pleae remove the body of the license. Just say it is UPL 1.0

File: OracleGoldenGate/runOracleGoldenGate.sh [hunk 9]
```diff
@@ -1,7 +1,33 @@
 #!/bin/bash
-# LICENSE CDDL 1.0 + GPL 2.0
+# Copyright (c) 2017 Oracle and/or its affiliates.  All rights reserved.
 #
-# Copyright (c) 2017 Oracle and/or its affiliates. All rights reserved.
+# The Universal Permissive License (UPL), Version 1.0
```
Comment (brunoborges) [inline OracleGoldenGate/runOracleGoldenGate.sh]: Please remove the body of the license. Just say it is UPL 1.0

**Pull request merged**